frontend/src/app/testQueue/TestCardForm/TestCardForm.tsx

```diff
@@ -0,0 +1,563 @@
+  useAOEFormOption,
+  useAppInsightTestCardEvents,
+  useDeviceTypeOptions,
+  useSpecimenTypeOptions,
+  useTestOrderPatient,
+} from "./TestCardForm.utils";
+import IncompleteAOEWarningModal from "./IncompleteAOEWarningModal";
+
+const DEBOUNCE_TIME = 300;
+
+export interface TestCardFormProps {
+  testOrder: QueriedTestOrder;
+  devicesMap: DevicesMap;
+  facility: QueriedFacility;
+  refetchQueue: () => void;
+  startTestPatientId: string | null;
+  setStartTestPatientId: React.Dispatch<React.SetStateAction<string | null>>;
+}
+
+const TestCardForm = ({
+  testOrder,
+  devicesMap,
+  facility,
+  refetchQueue,
+  startTestPatientId,
+  setStartTestPatientId,
+}: TestCardFormProps) => {
+  const initialFormState: TestFormState = {
+    dirty: false,
+    dateTested: testOrder.dateTested,
+    deviceId: testOrder.deviceType.internalId ?? "",
+    devicesMap: devicesMap,
+    specimenId: testOrder.specimenType.internalId ?? "",
+    testResults: convertFromMultiplexResponse(testOrder.results),
+    covidAOEResponses: {
+      pregnancy: testOrder.pregnancy as PregnancyCode,
+      noSymptoms: testOrder.noSymptoms,
+      symptomOnset: testOrder.symptomOnset,
+      symptoms: JSON.stringify(parseSymptoms(testOrder.symptoms)),
+    },
+  };
+  const [state, dispatch] = useReducer(testCardFormReducer, initialFormState);
+  const [useCurrentTime, setUseCurrentTime] = useState(!testOrder.dateTested);
+  const [hasAttemptedSubmit, setHasAttemptedSubmit] = useState(false);
+
+  const [editQueueItem, { loading: editQueueItemMutationLoading }] =
+    useEditQueueItemMutation();
+  const [updateAoeMutation, { loading: updateAoeMutationLoading }] =
+    useUpdateAoeMutation();
+  const [submitTestResult, { loading: submitLoading }] =
+    useSubmitQueueItemMutation();
+
+  const { trackSubmitTestResult, trackUpdateAoEResponse } =
+    useAppInsightTestCardEvents();
+
+  const submitModalRef = useRef<ModalRef>(null);
+
+  const { deviceTypeOptions, deviceTypeIsInvalid } = useDeviceTypeOptions(
+    facility,
+    state
+  );
+  const { specimenTypeOptions, specimenTypeIsInvalid } =
+    useSpecimenTypeOptions(state);
+
+  const { patientFullName } = useTestOrderPatient(testOrder);
+
+  const deviceSupportsMultiplex = doesDeviceSupportMultiplex(
+    state.deviceId,
+    devicesMap
+  );
+
+  const whichAOEFormOption = useAOEFormOption(state.deviceId, devicesMap);
+
+  /**
+   * When backend sends an updated test order, update the form state
+   * see refetch function and periodic polling on TestQueue useGetFacilityQueueQuery
+   */
+  useEffect(() => {
+    // don't update if there are unsaved dirty changes or if still awaiting saved edits
+    if (state.dirty || editQueueItemMutationLoading || updateAoeMutationLoading)
+      return;
+    dispatch({
+      type: TestFormActionCase.UPDATE_WITH_CHANGES_FROM_SERVER,
+      payload: testOrder,
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [testOrder]);
```

Comment:
I know I've had to disable this linting error before, but with how consistently we're ignoring it - is there a more correct way of doing this?  Or can we update the linter to not complain about this if its not actually an issue?

Reply:
Thanks for mentioning this! After I looked into it a bit more, the React team [strongly recommends keeping the rule](https://github.com/facebook/create-react-app/issues/6880#issuecomment-485912528) and reworking the code to handle it properly. I'm experimenting with the code and we may be able to remove some of these useEffects entirely, so I'll create a follow up PR for this.

Some of this effect code will benefit significantly when React officially releases the [new `useEffectEvent` hook](https://react.dev/learn/separating-events-from-effects#declaring-an-effect-event) on a stable version. For example, we need to control for race conditions between the frontend saving edits and the periodic live sync with any updates from the backend. 

When we get new values from the backend, we have to check whether the mutations are still loading and _although those loading booleans are reactive values_ that React says should be included in the dependency array, we don't want those in the array because then the effect will be triggered whenever those change and potentially cause sync issues.
```
useEffect(() => {
    // don't update if there are unsaved dirty changes or if still awaiting saved edits
    if (state.dirty || editQueueItemMutationLoading || updateAoeMutationLoading)
      return;
    dispatch({
      type: TestFormActionCase.UPDATE_WITH_CHANGES_FROM_SERVER,
      payload: testOrder,
    });
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [testOrder]);
```

So once `useEffectEvent` is officially added, we'll be able to separate reactive values that are actual effect dependencies from ones that are only used for conditional logic.